src/foam/dao/AbstractDAO.java

```diff
@@ -58,6 +58,8 @@ protected Sink decorateSink_(Sink sink, long skip, long limit, Comparator order,
       sink = new PredicatedSink(predicate, sink);
     }
 
+    ((AbstractFObject) sink).setX(x);
```

Comment:
I don't want to incur the extra overhead of calling x.create() for internal functionality where I may never need it. Especially now that the Context hasn't been heavily optimized yet. Also, I'm not sure if I want people to redefine underlying predicates like EQ for security reasons.